src/imap.rs

```diff
@@ -348,10 +348,11 @@ impl Imap {
                 connection_candidate,
             )
             .await
+            .context("IMAP failed to connect")
             {
                 Ok(client) => client,
                 Err(err) => {
-                    warn!(context, "IMAP failed to connect: {err:#}.");
+                    warn!(context, "{err:#}");
```

Comment:
Dot shouldn't be removed, errors don't contain dot at the end as far as i remember